pkg/envoy/proxy.go

```diff
@@ -205,7 +205,13 @@ func (e *envoy) Run(config interface{}, epoch int, abort <-chan error) error {
 }
 
 func (e *envoy) Cleanup(epoch int) {
-	filePath := configFile(e.Config.ConfigPath, epoch)
+	var filePath string
+	// when use the parameter "--templateFile=/path/xxx.tmpl"
+	if e.Config.CustomConfigFile != "" {
+		filePath = e.Config.CustomConfigFile
```

Comment:
I don't think we want to remove anything if its a custom file, only things we are generating

otherwise lgtm

Reply:
when use the parameter "--templateFile=/path/xxx.tmpl", It will  be generating a `envoy.yaml` file by pilot-agent. But cleanup is still `envoy-rev0.json`, not `envoy.yaml` file .

Reply:
oh I thought this would delete xxx.tmpl. Even then, if we delete envoy.yaml and a hot restart occurs it will break I think since now the config file is gone. That is why there is the `revX` added normally.

Reply:
Yeah，a warn level log will be printed, which does not look elegant, as follows. 
```
2020-03-10T03:04:02.209696Z	warn	Failed to delete config file /etc/istio/proxy/envoy-rev0.json for 0, remove /etc/istio/proxy/envoy-rev0.json: no such file or directory
```

Reply:
Why not just return early here then, instead of deleting the file?

Reply:
It has been modified and not be deleted in hot restart scenarios, thx.